compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -157,28 +157,35 @@ object ProtoTypes {
   abstract case class SelectionProto(name: Name, memberProto: Type, compat: Compatibility, privateOK: Boolean)
   extends CachedProxyType with ProtoType with ValueTypeOrProto {
 
-    /** Is the set of members of this type unknown? This is the case if:
-     *  1. The type has Nothing or Wildcard as a prefix or underlying type
-     *  2. The type has an uninstantiated TypeVar as a prefix or underlying type,
-     *  or as an upper bound of a prefix or underlying type.
+    /** Is the set of members of this type unknown, in the sense that we
+     *  cannot compute a non-trivial upper approximation? This is the case if:
+     *    1. The type has Nothing or Wildcard as a prefix or underlying type
+     *    2. The type is an abstract type with a lower bound that has a unknown
+     *       members and an upper bound that is both provisional and has unknown members.
+     *    3. The type is an uninstiated type var with a lower that has unknown members.
+     *    4. Type proxies have unknown members if their super types do
      */
-    private def hasUnknownMembers(tp: Type)(using Context): Boolean = tp match {
-      case tp: TypeVar => !tp.isInstantiated
+    private def hasUnknownMembers(tp: Type)(using Context): Boolean = tp match
       case tp: WildcardType => true
       case NoType => true
       case tp: TypeRef =>
         val sym = tp.symbol
-        sym == defn.NothingClass ||
-        !sym.isStatic && {
-          hasUnknownMembers(tp.prefix) || {
-            val bound = tp.info.hiBound
-            bound.isProvisional && hasUnknownMembers(bound)
-          }
-        }
+        sym == defn.NothingClass
+        ||     !sym.isClass
+            && !sym.isStatic
+            && {
+                hasUnknownMembers(tp.prefix)
+                || { val bound = tp.info.hiBound
+                     bound.isProvisional && hasUnknownMembers(bound)
+                  } && hasUnknownMembers(tp.info.loBound)
+              }
+      case tp: TypeVar =>
+        !tp.isInstantiated && hasUnknownMembers(TypeComparer.bounds(tp.origin).lo)
```

Comment:
Should this be:
```suggestion
      case tp: TypeVar if !tp.isInstantiated =>
        hasUnknownMembers(TypeComparer.bounds(tp.origin).lo)
```
Otherwise we won't recurse on the TypeVar instantiation which might itself be an uninstantiated TypeVar.

Reply:
I'm also wondering if this should be `TypeComparer.fullLowerBounds(tp.origin)`

Reply:
That's not what is was though. I am trying to be as conservative as possible since it seems to be really fickle here. Previously, uninstantiated type vars had unknown members, period. Now they only have them if theor lower bounds have unknown members. The proposed change would likely cause the same problems in concurrent.TreeMap that I have seen with the original changeset to `hasUnknownMembers`